src/mysql.erl

```diff
@@ -479,12 +479,15 @@ init(Opts) ->
     end,
 
     %% Connect socket
-    SockOpts = [{active, false}, binary, {packet, raw} | TcpOpts],
+    SockOpts = [binary, {packet, raw} | TcpOpts],
     {ok, Socket} = gen_tcp:connect(Host, Port, SockOpts),
 
     %% Exchange handshake communication.
     Result = mysql_protocol:handshake(User, Password, Database, gen_tcp,
                                       Socket),
+    inet:setopts(Socket, [{active, once}]),
+    MonitorRef = erlang:monitor(process, self()),
```

Comment:
Why monitor ourself? If we are dead, we won't get the 'DOWN' message anyway, do we?